src/core/createEnvForModule.ts

```diff
@@ -18,14 +18,14 @@ const createEnvForModule = (constantName: string) =>
     }, {});
 
 export const createAllModules = async (
-  modules: Record<string, ModuleFactory>,
+  modules: Record<string, ModuleCreator>,
 ): Promise<CreatedModule[]> => {
   const createdModules: CreatedModule[] = [];
 
-  for (const [name, factory] of Object.entries(modules)) {
+  for (const { name, factory } of Object.values(modules)) {
```

Comment:
I think we need a check here to avoid multiple module with the same name, WDYT ?

Reply:
Indeed, I'd suggest you create an issue for that and we deal with it later.